core/src/kmx/kmx_plus.cpp

```diff
@@ -158,6 +159,24 @@ COMP_KMXPLUS_KEYS::valid(KMX_DWORD _kmn_unused(length)) const {
   return true;
 }
 
+std::u16string
+COMP_KMXPLUS_KEYS_ENTRY::get_string() const {
+  if (flags & LDML_KEYS_FLAGS_EXTEND) {
+    return std::u16string();  // error, extended
+  }
+  char16_t buf[2];
+  std::size_t len;
+  if (Uni_IsBMP(to)) {
+    buf[0] = (to & 0xFFFF);
+    len    = 1;
+  } else {
+    buf[0] = Uni_UTF32ToSurrogate1(to);
+    buf[1] = Uni_UTF32ToSurrogate2(to);
```

Comment:
I'd love to rename these macros to something clearer also. When these were written, we were so Windows-specific that it was obvious that we were talking UTF-16. And surrogate implies UTF-16. But they should be `Uni_UTF32toUTF16surrogate1()`? or something like that ... or get away from macros and put all this into an (inline?) function, something like:

```c++
typedef struct { char16_t c[3]; } char16_single;
void Utf32CharToUtf16(const KMX_DWORD ch32, char16_single& ch16) { ... }
std::u16string Utf32CharToU16String(const KMX_DWORD ch32) { ... }
```

And I would null terminate `buf` for c-safety (so `char16_t buf[3]`). I am keen to try and improve our style as we move forward... so now is the time to do this!

Reply:
Ok

Reply:
done, and will rename macros in a separate PR